modules/graphics/src/shader.cxx

```diff
@@ -7,13 +7,19 @@ namespace Moon {
   glm::mat4 Shader::projection_matrix = glm::mat4(1.0f);
   glm::mat4 Shader::view_matrix = glm::mat4(1.0f);
 
-  Shader::Shader(const char* vertexfile, const char* fragmentfile)
+  Shader::Shader()
   {
-    m_program = CreateProgram(vertexfile, fragmentfile);
+    m_program = 0;
   };
 
   Shader::~Shader() {
-    glDeleteProgram(m_program);
+    if (m_program) {
+      glDeleteProgram(m_program);
+    }
+  };
+
+  void Shader::CreateFromString(const std::string vertexContent, const std::string fragmentContent) {
```

Comment:
CreateFromString, CreateFromFile can now be removed and CreateProgram can be moved to init.